File: src/main/java/com/dendrrahsrage/gui/hud/HUD.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.dendrrahsrage.gui.hud;

import com.jme3.app.Application;
import com.jme3.math.Vector3f;
import com.jme3.scene.Node;
import com.simsilica.lemur.Container;
import com.simsilica.lemur.Label;
import com.simsilica.lemur.ProgressBar;

public class HUD {

    public HUD(Application app, Node guiNode) {
        // Create a simple container for our elements
        Container topLeft = new Container();
        guiNode.attachChild(topLeft);

        // Put it somewhere that we will see it.
        // Note: Lemur GUI elements grow down from the upper left corner.
        topLeft.setLocalTranslation(0, app.getGuiViewPort().getCamera().getHeight(), 0);

        // Add some elements
        topLeft.addChild(new Label("D'endrrah's Rage v 0.0.0.0.0.0.0.0.0.1"));
        
        
        Container bottomLeft = new Container();
        guiNode.attachChild(bottomLeft);
        bottomLeft.setLocalTranslation(0, 32, 0);
        ProgressBar hp = new ProgressBar();
        bottomLeft.addChild(hp);
        hp.setProgressPercent(1);
        hp.setMessage("HP");
        hp.setPreferredSize(new Vector3f(200, 32, 1));
    }

}

File: src/main/java/com/dendrrahsrage/gui/hud/HUD.kt
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.dendrrahsrage.gui.hud

import com.dendrrahsrage.entity.EntityPlayer
import com.jme3.app.Application
import com.jme3.math.Vector3f
import com.jme3.scene.Node
import com.simsilica.lemur.Container
import com.simsilica.lemur.Label
import com.simsilica.lemur.ProgressBar

class HUD(
    app: Application,
    guiNode: Node,
    val player: EntityPlayer,
) {

    val weight = Label("Total weight")
    val location = Label("Location")
    val hp = ProgressBar()
    val hunger = ProgressBar()

    init {
        val topLeft = Container()
        guiNode.attachChild(topLeft)
        topLeft.setLocalTranslation(0f, app.guiViewPort.camera.height.toFloat(), 0f)
        topLeft.addChild(Label("D'endrrah's Rage v 0.0.0.0.0.0.0.0.0.1"))
        topLeft.addChild(location)

        val topRight = Container()
        guiNode.attachChild(topRight)
        topRight.setLocalTranslation(app.guiViewPort.camera.width.toFloat() - 64, app.guiViewPort.camera.height.toFloat(), 0f)
        topRight.addChild(weight)


        val bottomLeft = Container()
        guiNode.attachChild(bottomLeft)
        bottomLeft.setLocalTranslation(0f, 64f, 0f)
        bottomLeft.addChild(hp)
        hp.progressPercent = 1.0
        hp.message = "HP"
        hp.preferredSize = Vector3f(200f, 32f, 1f)
        bottomLeft.addChild(hunger)
        hunger.progressPercent = 1.0
        hunger.message = "Hunger"
        hunger.preferredSize = Vector3f(200f, 32f, 1f)
    }

    fun update() {
        hp.progressValue = player.betterPlayerControl.health
        hunger.progressValue = player.betterPlayerControl.hunger
        weight.text = player.betterPlayerControl.inventory.currentWeight.toString()+"kg"
        location.text = "${player.worldTranslation}"
    }
}
